Sources/MapboxMaps/MapView/Configuration/MapInitOptions.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+@objc public protocol MapInitOptionsDataSource {
+    /// When you implement this method you should return a `MapInitOptions`.
+    func mapInitOptions() -> Any
+}
+
+/// Options used when initializing `MapView`.
+///
+/// Contains the `ResourceOptions`, `MapOptions` (including `GlyphsRasterizationOptions`)
+/// that are required to initialize a `MapView`.
+public struct MapInitOptions: Equatable {
+
+    /// Associated `ResourceOptions`
+    public let resourceOptions: ResourceOptions
+
+    /// Associated `MapOptions`
+    public let mapOptions: MapOptions
```

Comment:
I wonder if we should provide an `initialStyleUrl` here similar to how Android provides an `initialCameraOptions`.. cc @pengdev

Reply:
Let's discuss!

Reply:
I think it's a good idea 👍 , and please also investigate if you can add the initialCameraOptions to the MapInitOptions on iOS. so that we are fully aligned.